src/main/java/StressMain.java

```diff
@@ -649,6 +655,54 @@ private static Callable taskCallable(Workflow workflow, SolrCloud cloud, Map<Str
 				} catch (Exception ex) {
 					ex.printStackTrace();
 				}
+			} else if (type.moveReplica != null) {
+				long taskStart = System.currentTimeMillis();
+
+				String collectionName = params.get("COLLECTION");
+				String replicaName = "", fromNodeName = "";
+				try (CloudSolrClient client = new CloudSolrClient.Builder(List.of(cloud.getZookeeperUrl()), Optional.ofNullable(cloud.getZookeeperChroot())).build();) {
+					ClusterState state = client.getClusterStateProvider().getClusterState();
+					DocCollection collection = state.getCollection(collectionName);
+					List<Replica> replicas = collection.getReplicas();
+					// choose first replica since it's easy and should exist
+					Replica r = replicas.get(0);
+					replicaName = r.getName();
+					fromNodeName = r.getNodeName();
+				} catch (Exception e) {
+					log.warn("Exception occurred getting replicaName and fromNodeName: " + e.getMessage());
+				}
+
+				String targetNodeName = "";
+				// pick random node to move to
+				List<SolrNode> copyOfSolrNodes = new ArrayList<>(cloud.nodes);
+				Collections.shuffle(copyOfSolrNodes);
+				for (SolrNode sn : copyOfSolrNodes) {
+					String toNodeName = sn.getNodeName() + "_solr";
+					if (toNodeName != fromNodeName) {
```

Comment:
minor nit - little bit safer to use string.equals

Reply:
Ah of course, bit rusty on my Java skills